new-log-viewer/src/utils/url.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Extracts the basename (filename) from a given URL string.
+ *
+ * @param urlString
+ * @return The extracted basename or "Unknown Filename" if extraction fails.
+ */
+const getBasenameFromUrl = (urlString: string): string => {
+    let basename = "Unknown Filename";
+    try {
+        const url = new URL(urlString);
+        const parts = url.pathname.split("/");
+        const popped = parts.pop();
+        if ("string" === typeof popped) {
```

Comment:
"undefined"

Reply:
How come?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/pop#return_value

pop() returns `undefined` when the array is empty. TypeScript is not smart enough to tell if an array is empty or not.

Since we're not logging `popped` being unexpectedly `undefined` when such happens, alternatively we can cast the return type in Line 16:
```
const popped = parts.pop() as string;
```

Reply:
Gotcha. Perhaps we should add a comment when we're using unnecessary code to satisfy the typescript compiler. On that note, I do prefer the cast you suggested.